clients/mobile/src/setup/notification/NotificationProvider.tsx

```diff
@@ -1,74 +1,43 @@
-import React, { useState, useEffect, ReactNode } from 'react'
+import React, { useState, useEffect, ReactNode, createContext } from 'react'
 import * as Notifications from 'expo-notifications'
 import { NavigationProp, useNavigation } from '@react-navigation/native'
 import { SCREENS } from '../constant/screens'
 import { parseJsonData } from '../../utility/jsonParser'
-import { NotificationContext } from './NotificationContext'
+import { NotificationContextType } from './useNotificationContext'
 import { useNavigationReady } from './useNavigationReady'
-import { NotificationData } from './NotificationData'
+import { NotificationData, NotificationDataTypes } from './NotificationData'
+import { RootStackParamList } from '../navigation/navigationTypes'
 
-type RootStackParamList = {
-  Home: undefined;
-  ResponseDonationRequest: { notificationData: NotificationData };
+const SCREEN_FOR_NOTIFICATION: Partial<Record<string, { screen: keyof RootStackParamList; getParams?: (data: Record<string, unknown>) => NotificationData }>> = {
+  bloodRequestPost: { screen: SCREENS.BLOOD_REQUEST_PREVIEW },
+  donorAcceptRequest: { screen: SCREENS.DONAR_RESPONSE, getParams: (data) => ({ notificationData: data }) }
 }
 
+export const initialNotificationState: NotificationDataTypes = {
+  notificationData: null
+}
+
+export const NotificationContext = createContext<NotificationContextType>(initialNotificationState)
+
 export const NotificationProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
   const [notificationData, setNotificationData] = useState<NotificationData | null>(null)
   const navigation = useNavigation<NavigationProp<RootStackParamList>>()
   const waitForNavigationReady = useNavigationReady(navigation)
 
   useEffect(() => {
     let isMounted = true
```

Comment:
i am not sure we need this

Reply:
This is necessary to avoid state update after unmount. Asynchronous `checkInitialNotification` function fetches notification and process as necessary. If the component is unmounted before this function completes, calling `navigation.navigate` creates an error. Mostly this happens when our app isn't running in background.

Reply:
I don't see it ever being reassigned